bvm/ballerina-core/src/main/java/org/ballerinalang/model/types/BRecordType.java

```diff
@@ -53,12 +54,28 @@ public TypeInfo getTypeInfo() {
 
     @Override
     public <V extends BValue> V getZeroValue() {
-        return null;
+        BMap<String, BValue> implicitInitValue = new BMap<>(this);
+        this.fields.entrySet().stream()
+                .filter(entry -> !Flags.isFlagOn(entry.getValue().flags, Flags.OPTIONAL))
+                .forEach(entry -> {
+                    BValue value = entry.getValue().fieldType.getZeroValue();
+                    implicitInitValue.put(entry.getKey(), value);
+                });
+
+        return (V) implicitInitValue;
     }
 
     @Override
     public <V extends BValue> V getEmptyValue() {
-        return (V) new BMap<>(this);
+        BMap<String, BValue> implicitInitValue = new BMap<>(this);
```

Comment:
can replace with `return getZeroValue()`

Reply:
Can't do it for record type since we are calling `getEmptyValue()` recursively on the fields. But can do this for some of the others like `BTableType`